From 7c0bd4c9dc1dd1ab352d4b0d534b555847fc9c54 Mon Sep 17 00:00:00 2001
From: Maya Lekova <mslekova@chromium.org>
Date: Mon, 26 Aug 2019 09:50:10 +0200
Subject: [PATCH] Merged: [turbofan] Fix memory corruption

Revision: f16a3a743641eaf8e3765272d3b1ad8eb44ee994

BUG=chromium:997057
LOG=N
NOTRY=true
NOPRESUBMIT=true
NOTREECHECKS=true
R=hablich@chromium.org

Change-Id: Ie052431da1481a16306cb7069e4b289f5bc21cf8
Reviewed-on: https://chromium-review.googlesource.com/c/v8/v8/+/1768363
Reviewed-by: Michael Hablich <hablich@chromium.org>
Cr-Commit-Position: refs/branch-heads/7.6@{#60}
Cr-Branched-From: 2cb25737c9eb7893ac752674e75d400b4419f67c-refs/heads/7.6.303@{#1}
Cr-Branched-From: 201c50951d68ef139c2229f707af0db57ce2318a-refs/heads/master@{#61902}
---
 src/compiler/js-native-context-specialization.cc | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/deps/v8/src/compiler/js-native-context-specialization.cc b/deps/v8/src/compiler/js-native-context-specialization.cc
index 312ab38f517..be4c111b265 100644
--- a/deps/v8/src/compiler/js-native-context-specialization.cc
+++ b/deps/v8/src/compiler/js-native-context-specialization.cc
@@ -1059,7 +1059,8 @@ Reduction JSNativeContextSpecialization::ReduceNamedAccess(
   Node* control = NodeProperties::GetControlInput(node);
 
   ZoneVector<PropertyAccessInfo> access_infos(zone());
-  AccessInfoFactory access_info_factory(broker(), dependencies(), zone());
+  AccessInfoFactory access_info_factory(broker(), dependencies(),
+                                        graph()->zone());
   if (!access_info_factory.FinalizePropertyAccessInfos(
           feedback.access_infos(), access_mode, &access_infos)) {
     return NoChange();
@@ -1795,7 +1796,7 @@ JSNativeContextSpecialization::ReducePropertyAccessUsingProcessedFeedback(
         if (name.has_value()) {
           ZoneVector<PropertyAccessInfo> access_infos(zone());
           AccessInfoFactory access_info_factory(broker(), dependencies(),
-                                                zone());
+                                                graph()->zone());
           access_info_factory.ComputePropertyAccessInfos(
               receiver_maps, name->object(), access_mode, &access_infos);
           processed = new (zone()) NamedAccessFeedback(*name, access_infos);
